sdk/cosmosdb/azure_data_cosmos/examples/cosmosdb_connect.rs

```diff
@@ -0,0 +1,45 @@
+use azure_data_cosmos::{CosmosClient, CosmosClientMethods, DatabaseClientMethods};
+use clap::Parser;
+
+/// A simple example to show connecting to a Cosmos Account and retrieving the properties of a database.
+#[derive(Parser)]
+pub struct Args {
+    /// The CosmosDB endpoint to connect to.
+    endpoint: String,
+
+    /// The database to fetch information for.
+    database: String,
+
+    /// An authentication key to use when connecting to the Cosmos DB account. If omitted, the connection will use Entra ID.
+    #[clap(long)]
+    #[cfg(feature = "key-auth")]
```

Comment:
Seems a bit complicated to add feature support to an example. Since you have to check for `Some` anyway, maybe just require the `key_auth` (nee `key-auth`) feature? Granted, that's not a great UX either since callers have to explicitly pass it (which `[[example]]` sections could just define what features to enable, but alas).